api/internal/localizer/localizer.go

```diff
@@ -75,9 +76,69 @@ func (lc *Localizer) Localize() error {
 	return nil
 }
 
-// TODO(annasong): implement
-// processKust returns a copy of the kustomization at kt with paths localized.
-func (lc *Localizer) processKust(kt *target.KustTarget) *types.Kustomization {
+// processKust returns a copy of the kustomization at kt with the patches field localized
+func (lc *Localizer) processKust(kt *target.KustTarget) (*types.Kustomization, error) {
 	kust := kt.Kustomization()
-	return &kust
+	for name, patches := range map[string][]types.Patch{
+		"patches":         kust.Patches,
+		"patchesJson6902": kust.PatchesJson6902,
+	} {
+		for i := range patches {
+			if patches[i].Path != "" {
+				newPath, err := lc.localizeFile(patches[i].Path)
+				if err != nil {
+					return nil, errors.WrapPrefixf(err, "unable to localize %s path", name)
+				}
+				patches[i].Path = newPath
+			}
+		}
+	}
+	return &kust, nil
+}
+
+// localizeFile localizes file path and returns the localized path
+func (lc *Localizer) localizeFile(path string) (string, error) {
+	content, err := lc.ldr.Load(path)
+	if err != nil {
+		return "", errors.Wrap(err)
+	}
+
+	var locPath string
+	if loader.IsRemoteFile(path) {
+		if !lc.addLocalizeDir() {
+			return "", errors.Errorf("cannot localize remote %q: localize directory already exists at %q", path,
+				lc.ldr.Root())
+		}
```

Comment:
This validation check is not implemented. Can you use a "TODO" comment? 
```
if loader.IsRemoteFile(path) {
   // TODO: ... 
   locPath = locFilePath(path)
```
Reason: This mainly affects how unittest should be added. if you add the validation check in this PR, we should have the unittest to cover cases where localdir cannot be added. But since `addLocalizeDir` is not implemented, you cannot really validate that condition. Once you have  `addLocalizeDir` implemented, it would also make sense to include where the function is called in the same PR.

Reply:
Good point. And thanks for the detailed explanation!